mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js

```diff
@@ -0,0 +1,144 @@
+    const [isOtherChecked, setIsOtherChecked] = useState(!!value?.otherValue)
+
+    const _loadNonOtherCheckboxItems = useEffect(() => {
+      setNonOtherCheckboxItems(value?.selectedValues ?? [])
+    }, [value?.selectedValues])
+
+    const handleOtherCheckboxChange = () => {
+      const isOtherCheckedNewState = !isOtherChecked
+      setIsOtherChecked(isOtherCheckedNewState)
+      onChange(
+        getValueToReturn({
+          nonOtherCheckboxItems,
+          otherInputValue,
+          isOtherChecked: isOtherCheckedNewState
+        })
+      )
+    }
+    const handleOtherTextInputChange = (event) => {
+      const {
+        target: { value }
+      } = event
+      setOtherInputValue(value)
+      onChange(
+        getValueToReturn({
+          nonOtherCheckboxItems,
+          otherInputValue: value,
+          isOtherChecked
+        })
+      )
+    }
+    const handleNonOtherCheckboxChange = (itemValue) => {
+      const updateNonOtherCheckboxItems = [...nonOtherCheckboxItems]
+      const foundItemIndex = updateNonOtherCheckboxItems.indexOf(itemValue)
+
+      if (foundItemIndex > -1) {
+        updateNonOtherCheckboxItems.splice(foundItemIndex, 1)
+      } else {
+        updateNonOtherCheckboxItems.push(itemValue)
+      }
+
+      setNonOtherCheckboxItems(updateNonOtherCheckboxItems)
+      onChange(
+        getValueToReturn({
+          nonOtherCheckboxItems: updateNonOtherCheckboxItems,
+          otherInputValue,
+          isOtherChecked
+        })
+      )
+    }
+
+    const otherThanOtherCheckboxInputs = options.map((item) => (
+      <div key={item.value}>
+        <FormControlLabel
+          control={
+            <Checkbox
+              id={`${id}-${item.value}`}
+              value={item.value}
+              checked={nonOtherCheckboxItems.includes(item.value)}
+              onChange={() => {
+                handleNonOtherCheckboxChange(item.value)
+              }}
+              onBlur={onBlur}
+            />
+          }
+          label={item.label}
+        />
+      </div>
+    ))
+
+    return (
+      <FormGroup ref={ref}>
+        {otherThanOtherCheckboxInputs}
+        {shouldAddOtherOptionWithClarification ? (
+          <div>
+            <FormControlLabel
+              control={
+                <Checkbox
+                  id={`${id}-other`}
+                  value={'OTHER'}
+                  checked={isOtherChecked}
+                  onChange={handleOtherCheckboxChange}
+                  onBlur={onBlur}
+                />
+              }
+              label={language.form.checkboxGroupOtherLabel}
+            />
+            {isOtherChecked ? (
+              <TextField
+                id='outlined-basic'
+                label={language.form.checkboxGroupOtherInputPlaceholder}
+                variant='outlined'
+                onChange={handleOtherTextInputChange}
+                value={otherInputValue}
+              />
+            ) : null}
```

Comment:
you don't need this ternary in jsx, you just do `isOtherChecked && ( < jsx ....`